Pages as pipeline

roopa-pai opened the pull request:
related to issue #63 

Structured website pages in the form of a neuroscience/neuroimaging pipeline. Created top-level pages. Added links in new pages to the existing, content-containing *.md docs/pages.

roopa-pai committed:
Add first 7 top-level docs.

roopa-pai committed:
added remaining top-level files

roopa-pai committed:
added links to existing content-containing pages in top-level docs.

roopa-pai committed:
Fixed broken internal links. removed spaces; added dashes between header-words.

roopa-pai committed:
bullet-point formatting

roopa-pai committed:
Merge pull request #8 from roopa-pai/roopa-CI

created files: .remarkrc, npm-requirements.txt, .travis.yml. (Ref: ht…

roopa-pai committed:
Merge pull request #10 from roopa-pai/master

merge CI part into pages-as-pipeline

roopa-pai committed:
update mkdocs.yml to add "pipeline" structure to nav. added link to "open-sciene.md" in "08_Sharing-data-code-and-results.md".

roopa-pai committed:
fixed multilevel nav in "mkdocs.yml"

roopa-pai committed:
fixed broken link in nav in "mkdocs.yml"

Remi-Gau commented on mkdocs.yml:
```diff
@@ -15,15 +15,31 @@ theme:
-  - Reproducible research: "reproducible_neuroimaging_research.md"
+    - Learning neuroimaging: "learning_neuroimaging.md"
+    - Neuroimaging resources: "neuroimaging_resources.md"
+  - Quality control:
```
Maybe I should have realized earlier but maybe quality control should come after the analysis, no?

If we said that we want to have QC as a separate section maybe it should come at the very end (before `reporting`)`and then we create those internal links to it.

roopa-pai replied:
Hm, personally it makes intuitive sense to me for QC to be where it is (after processing and before analysis), but if you think it should be after analysis I will defer to you! I will shuffle it to the end.

Remi-Gau replied:
if you don't mind it would make me happy. :-)

Remi-Gau commented on docs/02_Planning-your-study.md:
```diff
@@ -0,0 +1,27 @@
+
+* [PLANNING YOUR STUDY]
+
+ Resources currently collected [here](/docs/reproducible_neuroimaging_research.md#Reproducible-neuroimaging-research)
```
apparently I get a 404 error when I try this link (and with a few others)

The only way I have been able to make this work is with:

 `Resources currently collected [here](./reproducible_neuroimaging_research.md#reproducible-neuroimaging-tools)`

roopa-pai replied:
I will fix this! Thanks for spotting it. Some links seem to "work" in atom, but don't on the "mkdocs serve"ed website.

Remi-Gau commented on docs/01_Before-you-start.md:
```diff
@@ -0,0 +1,15 @@
+
+Before you start your study, here are some things you might want to consider:
+
+* [Reproducibility]
```
I was thinking that until we have filled those subsections we could comment out those, no?

This way we have our structure in the markdown files but it is not displayed in the mkdocs until we put something in that section.

Alternatively we could keep those sections to show people what is our goal structure.

```
<!-- * [Reusing data]
* [Defining your terms and your task]
* [Ontologies]
* [Piloting]
* [Pre-registration]
* [Optimizing your design]
* [Design efficiency]
* [Power]
* [For MVPA: same analysis approach]
* [Defining your region of interest]
* [Using previous results]
* [Localizers]
* [Atlases]
* [Non-standard templates]
* [Stimuli presentation softwares]
 * [psychopy]
 * [expyriment]
 * [psychtoolbox]
* [Existing protocols] -->
```

roopa-pai replied:
Sounds good!

roopa-pai replied:
Ah no worries, I can do it! I can use the learning experience :)

Remi-Gau commented:
By the way I could help you also directly push on this merge request so we both work on it, but I would prefer to avoid merge conflicts. 
So if you want to split the work, let me know and I can take care of some files: just tell me which ones. :-)

roopa-pai commented:
> By the way I could help you also directly push on this merge request so we both work on it, but I would prefer to avoid merge conflicts.
> So if you want to split the work, let me know and I can take care of some files: just tell me which ones. :-)

Ah no worries, I can do it! I can use the learning experience :)

roopa-pai committed:
commented out list of resource-topics on some pages(to be added later). Moved QC section to after Analysis section.

roopa-pai committed:
commenting fixed: added newlines before and after comment-text.

roopa-pai committed:
edited mkdocs.yml: fixed nav indentation for QC.

roopa-pai commented:
@Remi-Gau done! Let me know if things work as they should for you. For me it seems to be fine.

Remi-Gau commented:
Started checking. Made me realize that there should be one thing that should be "automatable" by travis: **checking for dead links**. :wink:

roopa-pai commented:
I was thinking the same! I know it's a "rule" in the linter

Remi-Gau committed:
fix linter issues and comments

- start fixing some linter issues (e.g no _ in filenames)
- some commented out section still showed up in the doc

Remi-Gau committed:
update mkdocs.yml and fix linter issues

Remi-Gau committed:
Merge branch 'master' into pages-as-pipeline

Remi-Gau committed:
fix more linter issues

Remi-Gau commented:
I have fixed some parts that were commented out but still showed in the rendered MKDOCS (not sure why).

And I started to do some batch fixing of the linter issues with some find and replace.

Maybe have a look that you are OK with it. If so feel free to merge it.

roopa-pai reviewed (commented):
all good!

roopa-pai commented:
> I have fixed some parts that were commented out but still showed in the rendered MKDOCS (not sure why).
> 
> And I started to do some batch fixing of the linter issues with some find and replace.
> 
> Maybe have a look that you are OK with it. If so feel free to merge it.

Just saw this, thanks for the edits. I'll merge it now!

Remi-Gau reviewed (approved):
let's merge this baby

Remi-Gau commented:
@roopa-pai I will let you do the honor of merging it. :-)
